packages/client/components/SpotlightModal.tsx

```diff
@@ -183,8 +186,28 @@ const SpotlightModal = (props: Props) => {
   const {viewer} = data
   const {meeting} = viewer
   const phaseRef = useRef(null)
-  if (!meeting) return null
-  const {spotlightReflection} = meeting
+  const spotlightReflectionIds = useRef<null | string[]>(null)
+  const spotlightGroup = meeting?.spotlightGroup
+  const firstReflectionId = spotlightGroup?.reflections[0]?.id
+  // const secondReflectionId = spotlightGroup?.reflections[1]?.id
+  const showSpotlight = meeting && spotlightGroup
+
+  useEffect(() => {
+    if (!spotlightGroup) return
+    const {current: ids} = spotlightReflectionIds
+    if (!ids && firstReflectionId) {
+      spotlightReflectionIds.current = [firstReflectionId]
+    }
+    // TODO: uncomment for groups -> source issue
+    // else if (firstReflectionId && secondReflectionId && ids?.includes(secondReflectionId)) {
+    //   spotlightReflectionIds.current = [...ids, firstReflectionId]
+    // }
+    else if (!firstReflectionId || !ids?.includes(firstReflectionId)) {
+      setTimeout(() => {
+        closeSpotlight()
+      }, Times.REFLECTION_DROP_DURATION)
```

Comment:
Instead of `setTimeout`, we could get the reflection that was in the Spotlight, find that reflection in the meeting reflections, wait until `isDropping` is false, and then close the Spotlight. I like to avoid `setTimeout`s wherever possible but I think this is prob easier to understand and more performant.